src/NLog/Targets/Wrappers/LimitingTargetWrapper.cs

```diff
@@ -0,0 +1,174 @@
+        /// </summary>
+        public LimitingTargetWrapper():this(null)
+        {
+        }
+
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LimitingTargetWrapper" /> class.
+        /// </summary>
+        /// <param name="name">The name of the target.</param>
+        /// <param name="wrappedTarget">The wrapped target.</param>
+        public LimitingTargetWrapper(string name, Target wrappedTarget) 
+            : this(wrappedTarget, 1000, TimeSpan.FromHours(1))
+        {
+            this.Name = name;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LimitingTargetWrapper" /> class.
+        /// </summary>
+        /// <param name="wrappedTarget">The wrapped target.</param>
+        public LimitingTargetWrapper(Target wrappedTarget)
+            : this(wrappedTarget, 1000, TimeSpan.FromHours(1))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LimitingTargetWrapper" /> class.
+        /// </summary>
+        /// <param name="wrappedTarget">The wrapped target.</param>
+        /// <param name="messageLimit">Maximum number of messages written per interval.</param>
+        /// <param name="interval">Interval in which the maximum number of messages can be written.</param>
+        public LimitingTargetWrapper(Target wrappedTarget, int messageLimit, TimeSpan interval)
+        {
+            this.MessageLimit = messageLimit;
+            this.Interval = interval;
+            this.WrappedTarget = wrappedTarget;
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum allowed number of messages written per <see cref="Interval"/>.
+        /// </summary>
+        /// <remarks>
+        /// Messages received after <see cref="MessageLimit"/> has been reached in the current <see cref="Interval"/> will be discarded.
+        /// </remarks>
+        [DefaultValue(1000)]
+        public int MessageLimit { get; set; }
+
+        /// <summary>
+        /// Gets or sets the interval in which messages will be written up to the <see cref="MessageLimit"/> number of messages.
+        /// </summary>
+        /// <remarks>
+        /// Messages received after <see cref="MessageLimit"/> has been reached in the current <see cref="Interval"/> will be discarded.
+        /// </remarks>
+        [DefaultValue(typeof(TimeSpan), "01:00")]
+        public TimeSpan Interval { get; set; }
+
+        /// <summary>
+        /// Gets the <c>DateTime</c> when the current <see cref="Interval"/> will be reset.
+        /// </summary>
+        public DateTime IntervalResetsAt { get { return firstWriteInInterval + Interval; } }
+
+        /// <summary>
+        /// Gets the number of <see cref="AsyncLogEventInfo"/> written in the current <see cref="Interval"/>.
+        /// </summary>
+        public int MessagesWrittenCount { get { return writeCount;} }
```

Comment:
I prefer a `{get;private set}` property and removal of the field. (less = more).